### PR TITLE
1826 su user queries

### PR DESCRIPTION
**Pull request by jordanh**
# Test Cycle Template
## Tests

For all tests, open `/admin/graphql`

### renamed query
- [ ] `suCountTiersForUser(userId: "auth0|5797e83170dddc395d8d1675")` returns which tiers seed user belongs to

### renamed query with updated params:
- [ ] query for `suOrgCount` returns list of active PRO orgs
- [ ] query for `suOrgCount(includeInactive: true)` returns number of all active _and_ inactive PRO orgs
- [ ] query for `suOrgCount(tier: personal)` returns number of all active PERSONAL orgs
- [ ] query for `suUserCount` returns number of active PRO users
- [ ] query for `suUserCount(includeInactive: true)` returns number of all active _and_ inactive PRO users
- [ ] query for `suUserCount(tier: personal)` returns number of all active PERSONAL users
 
### new query:
Running this query...
```
query {
  suProOrgInfo {
    activeCount,
    organization {
      billingLeaders {
        email
      }
    }
  }
}
```

- [ ] results in specified fields resolved correctly

## Test notes

### CODE REVIEW

**Comment by jordanh** [comment]
@mattkrick this is ready for review. Nothing big here. I don't expect to use `suProOrgInfo` for anything other than sanity checking. Most of this info should now be available to us in HubSpot.

File: package.json [hunk 1]
```diff
@@ -42,9 +42,10 @@
     "flow": "flow",
     "test": "jest --bail --coverage --ci --runInBand && codecov",
     "test:dev": "jest --runInBand --watch",
-    "test:intranetToken": "node ./src/server/utils/getIntranetToken.babel.js",
-    "test:e2e-deps": "node ./e2e/scripts/ensureDrivers.babel.js",
     "test:e2e": "node ./e2e/scripts/runTests.babel.js",
+    "test:e2e-deps": "node ./e2e/scripts/ensureDrivers.babel.js",
+    "test:intranetToken": "node ./src/server/utils/getIntranetToken.babel.js",
+    "test:update": "jest --updateSnapshot --runInBand",
```
Comment (jordanh) [inline package.json]: I've needed this before. Safe to add `test:update`?

File: src/server/graphql/queries/suProOrgCount.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    return r.table('Organization')
+      .getAll(PRO, {index: 'tier'})
+      .map((org) => r.branch(includeInactive,
+        true, // count all orgs
+        org('orgUsers')
+        // calculate whether the org is active or not:
+        //   reduces [bool, bool, bool] => bool
+        //   where at least one false => true (active org)
+        //   all true => false (inactive org)
+          .fold(false, (acc, orgUser) => acc.or(r.not(orgUser('inactive'))))
```
Comment (jordanh) [inline src/server/graphql/queries/suProOrgCount.js]: When I calculate these metrics including the 0-count Pro orgs, @tayamueller didn't like it. We can now use this more pragmatic query parametrically.

**Comment by jordanh** [comment]
@mattkrick after collecting the metrics this evening I was compelled to do some late night refactoring. I wanted to be able to collect counts from non-Pro tiers using the same queries. So, I adjusted some things.

**Comment by jordanh** [comment]
@mattkrick ok, I think these are ready to go now

File: package.json (hunk 1 above)
Comment (jordanh) [reply package.json]: Ok, I will pull this line.

File: src/server/graphql/queries/__tests__/activeProOrgCount.test.js [hunk 3]
```diff
@@ -1,44 +0,0 @@
...
-  // SETUP
-  const mockDB = new MockDB();
-  const {user} = await mockDB.init();
-  const authToken = mockAuthToken(user[1], {rol: 'su'});
-  // TEST
-  await mockDB.newOrg({tier: PRO});
-  const next = await activeProOrgCount.resolve(undefined, undefined, {authToken});
-
-  // VERIFY
-  expect(next >= 1).toBe(true);
```
Comment (mattkrick) [inline src/server/graphql/queries/__tests__/activeProOrgCount.test.js]: this was the cheapest way i figured i could rewrite these tests to support parallel testing

File: src/server/graphql/queries/__tests__/suOrgCount.test.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  const mockDB = new MockDB();
+  const {user} = await mockDB.init();
+  const authToken = mockAuthToken(user[1], {rol: 'su'});
+  // TEST
+  const initial = await suOrgCount.resolve(undefined, defaultResolverArgs, {authToken});
+  await mockDB.newOrg({tier: PRO});
+  const next = await suOrgCount.resolve(undefined, defaultResolverArgs, {authToken});
+
+  // VERIFY
+  expect(next - initial).toEqual(1);
```
Comment (mattkrick) [inline src/server/graphql/queries/__tests__/suOrgCount.test.js]: comparing next to initial won't work for running tests in parallel.

File: package.json (hunk 1 above)
Comment (jordanh) [reply package.json]: I captured this bit of knowledge for time immemorial: https://app.getguru.com/#/facts/038c60ea-0b72-4a77-a2f3-98dbd4d870f0

File: src/server/graphql/queries/__tests__/activeProOrgCount.test.js (hunk 3 above)
Comment (jordanh) [reply src/server/graphql/queries/__tests__/activeProOrgCount.test.js]: Ok, I made one small change to check for changes in the count instead (`!==`). Not ideal, but better. We'll move to another strategy in the future if we need to.

**Comment by jordanh** [comment]
@mattkrick ok, 👀 again pls

**Comment by mattkrick** [comment]
:+1:

**Pull request merged**